### PR TITLE
Alembic DB migrations

### LINKED ISSUE #83: Handle db migration?
Hello,

I'd like to try and add a feature to this app, that requires an sql schema migration.
Do you have an opinion about the best way to properly handle schema migrations?

Basically, all tools work the same and rely on a version field (integer) stored in a special table.
If not found, table and field are created (and version assumed to be 0).
If found, application compares the field with its own schema version, and if necessary, applies required migrations.

SqlAlchemy has its own tool: http://readthedocs.org/docs/sqlalchemy-migrate
They describe the generic steps involved: http://caneypuggies.alwaysreformed.com/wiki/DevelopingWithMigrations

Googling for "flask migration", I also found https://github.com/adamrt/flask-evolution/tree/master/flaskext -however I'm not sure what it is good for.

### PR DESCRIPTION
**Pull request by JocelynDelalande**
fix #83 

I'm a bit frustrated with sqlalchemy migrations compared to django ones, but well… Flask-Migrations + Alembic seems the suckless solution at the moment.

### CODE REVIEW

**Comment by JocelynDelalande** [comment]
@almet gentle bump :-) that's a blocker for #131 (that I will need to rewrite to take db migrations into account).

**Comment by almet** [comment]
I have almost no experience with migrations for Flask, so let's consider this is the better solution right now :-)

File: budget/migrations/env.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+from __future__ import with_statement
+from alembic import context
+from sqlalchemy import engine_from_config, pool
+from logging.config import fileConfig
+import logging
+
+# this is the Alembic Config object, which provides
+# access to the values within the .ini file in use.
```
Comment (almet) [inline budget/migrations/env.py]: nit: please start comments with a capital and end them with a dot. Wrap them at 80 chars (not less).

File: budget/migrations/env.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+config = context.config
+
+# Interpret the config file for Python logging.
+# This line sets up loggers basically.
+fileConfig(config.config_file_name)
+logger = logging.getLogger('alembic.env')
+
+# add your model's MetaData object here
```
Comment (almet) [inline budget/migrations/env.py]: nit: ditto

**Comment by almet** [comment]
Cool, this looks good to me apart from the nitpicks. Thanks for tackling this.

**Comment by JocelynDelalande** [comment]
@almet thanks for feedback :-).

I nitfixed (note: that was default/generated code from *flask-Migrations*), is it better like that ?

**Comment by almet** [comment]
This is now in master.

**Pull request merged**